### PR TITLE
Utility shuffling and tweaks

### PR DESCRIPTION
**Pull request by daviddrysdale**
Tweaking and moving utility functions around in preparation for re-use.

Best reviewed commit-by-commit.

### CODE REVIEW

File: x509util/files.go [hunk 1]
```diff
@@ -74,6 +75,21 @@ func dePEM(data []byte, blockname string) [][]byte {
 	return results
 }
 
+// ReadFileOrURL returns the data from a target which may be either a filename
+// or a URL.
```
Comment (Martin2112) [inline x509util/files.go]: "or an HTTP(S) URL" ?

File: scanner/scanner.go [hunk 2]
```diff
@@ -79,8 +79,9 @@ type Scanner struct {
 	// Configuration options for this Scanner instance
 	opts ScannerOptions
 
-	// Counter of the number of certificates scanned
+	// Counter of the number of certificates scanned and matched
```
Comment (Martin2112) [inline scanner/scanner.go]: nit: Counters or Counts.

Comment (daviddrysdale) [reply scanner/scanner.go]: Done.

File: x509util/files.go (hunk 1 above)
Comment (daviddrysdale) [reply x509util/files.go]: Done.

**Pull request merged**